src/JPEGView/WEBPWrapper.cpp

```diff
@@ -0,0 +1,135 @@
+#include "stdafx.h"
+
+#include "WEBPWrapper.h"
+#include "webp/decode.h"
+#include "webp/encode.h"
+#include "webp/demux.h"
+#include "MaxImageDef.h"
+#include "Helpers.h"
+
+struct WebpReaderWriter::webp_cache {
+	WebPAnimDecoder* decoder;
+	WebPData data;
+	int prev_frame_timestamp;
+	int width;
+	int height;
+};
+
+WebpReaderWriter::webp_cache WebpReaderWriter::cache = { 0 };
+
+void* WebpReaderWriter::ReadImage(int& width,
+	int& height,
+	int& nchannels,
+	bool& has_animation,
+	int& frame_count,
+	int& frame_time,
+	bool& outOfMemory,
+	const void* buffer,
+	int sizebytes)
+{
+	uint8* pPixelData = NULL;
+	WebPBitstreamFeatures features;
+	width = height = 0;
+	nchannels = 4;
+	outOfMemory = false;
+	if (!cache.decoder || !cache.data.bytes) {
+		if (!WebPGetInfo((const uint8_t*)buffer, sizebytes, &width, &height))
+			return NULL;
+		if (width > MAX_IMAGE_DIMENSION || height > MAX_IMAGE_DIMENSION)
+			return NULL;
```

Comment:
in the old code this was flagged as outOfMemory... should it set that here?  I know after all it was just a flag after all to indicate things were too big

Reply:
outOfMemory shouldn't be set when MAX_IMAGE_DIMENSION is exceeded since it's inaccurate (["must not be bigger than 65535 due to internal limitations"](https://github.com/sylikc/jpegview/blob/2082bd9f1d7864eaf36e6ec7062dd66386eca715/src/JPEGView/MaxImageDef.h#L47)), but maybe there should be another error message for it.

Reply:
ok, got it 👍